tests/std/tests/P0218R1_filesystem/test.cpp

```diff
@@ -1216,10 +1214,10 @@ void test_directory_entry() {
     for (auto&& nonexistent : nonexistentPaths) {
         cachingEntry.assign(nonexistent); // no fail

```

Comment:
Observation: The comment above this loop, `// assert that mutating the path doesn't fail even though the target doesn't exist`, and the `// no fail` comment for the non-`error_code` overload, strongly indicate that these functions should be succeeding - which seems to indicate that `refresh` should succeed for non-existent paths.